test/test_unary_ufuncs.py

```diff
@@ -215,6 +215,10 @@ def _fn(t):
             if alt is None:
                 continue
 
+            if not callable(alt):
```

Comment:
Should we just explicitly set the method_variant of real and imag to None instead of adding this check to the tests?

Reply:
That makes sense, as this behavior will be reflected by OpInfo object.

However how do you suggest we proceed about it. Two options pop into my mind without having to update the OpInfo base class itself.

Option 1:
```python

real_op_info = OpInfo(...)
real_op_info.method_variant = None

...
...

# Add object to OpInfo DB
op_db = [
...
real_op_info,
...
]
```

Option 2: Inherit from OpInfo and override `get_method`
```python

class AttributeInfo(OpInfo):
   def __init__():
       .....

   def get_method():
      return None
```

I prefer the second one. Let me know how you feel about it or if I missed some other option. 

Thanks!

Reply:
I would update the base class to directly accept the method variant just like it allows override of the function variant.

Reply:
Personally I'll prefer if it is explicitly visible in the form of field or some different class which explicitly screams that this is an `attribute`. What do you think?

Reply:
I guess my thinking is that the OpInfo is testing the real() and imag() functions and not testing the .real and .imag attributes, and the logic to acquire the method variant of real() is just wrong currently because there is no method variant of real and it's acquiring an attribute, .real, which we're not testing.

Does that make sense?

Reply:
Right. That is the reason it fails. But I was thinking OpInfo is like an entry which sort of captures the behaviour of an operator. So thought it'd be nice if there is an explicit way to mention, this is an attribute. 

But sure we can move callable to the point where we set the method variant.

Reply:
I agree with you. To merge our thinking we could have the OpInfo, and then we could create a new AttributeInfo class, like you're proposing, and AttributeInfo would be for .real and .imag, so there'd be an OpInfo for torch.real and torch.imag, and an AttributeInfo for torch.Tensor.real and torch.Tensor.imag. I'm not sure how useful an AttributeInfo would be, however, because the test functions we've written aren't designed for attributes.

Reply:
What I had in mind was something like this

```python
class AttributeOpInfo(UnaryUfuncInfo):
    '''
    OpInfo for attributes like `real` and `imag`
    Inherits from UnaryUfunc as attributes will be unary functions
    '''
    def get_method(self):
        return None

    def get_inplace(self):
        return None

....
....

# OpInfo Entry as usual
AttributeOpInfo('real',
                ref=np.real,
                dtypes=complex_types(),
                dtypesIfCPU=complex_types(),
                dtypesIfCUDA=complex_types(),
                dtypesIfROCM=complex_types(),
                supports_tensor_out=False,
                test_inplace_grad=False),

```

So by looking at it one can reason about it being an attribute.

Reply:
The only interesting tests I can think for tensor attributes are view tests. So, I think it might make more sense to add a `ViewInfo` and then do as @mruberry suggested:
` checks if the method variant is callable after acquiring it and, if not callable, it sets the method variant to None`. These `ViewInfo`s could then test all views and attributes!

Reply:
The challenge I have the proposed AttributeInfo is that it's testing the torch.real() op, not the `real` attribute of tensors, so it seems misleading?

I guess I'd expect AttributeInfos to not derive from OpInfos and just test actual tensor attributes using, for example, getattr in their tests.